src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/InitializeStaticFieldsInline.cs

```diff
@@ -92,5 +99,33 @@ public override void Initialize(AnalysisContext analysisContext)
                 });
             });
         }
+
+        private class AtomicBoolean
```

Comment:
I find reasoning with `Interlocked.CompareExchange` not so easy/readable hence this helper. Let me know if you want me to get rid of the helper or move it as a globally available helper.

Reply:
Lets leave it here for now. If we need to share in future, it can be moved.